server/src/grpc/remote_engine_service/mod.rs

```diff
@@ -331,21 +336,28 @@ async fn handle_stream_read(
 
     let begin = Instant::now();
     let table = find_table_by_identifier(&ctx, &table_ident)?;
-    let streams = table
+    let res = table
         .partitioned_read(read_request)
         .await
         .box_err()
         .with_context(|| ErrWithCause {
             code: StatusCode::Internal,
             msg: format!("fail to read table, table:{table_ident:?}"),
-        })?;
-
-    info!(
+        });
+    match res {
+        Ok(streams) => {
+            info!(
         "Handle stream read success, request_id:{request_id}, table:{table_ident:?}, cost:{:?}",
         begin.elapsed(),
     );
-
-    Ok(streams)
+            REMOTE_ENGINE_GRPC_HANDLER_COUNTER_VEC.query_succeeded.inc();
```

Comment:
How about adding a new type for this case? `stream_query_succeed` and `stream_query_failed`?